qadence/ml_tools/printing.py

```diff
@@ -20,5 +36,112 @@ def write_tensorboard(
         writer.add_scalar(key, arg, iteration)
 
 
-def log_hyperparams(writer: SummaryWriter, hyperparams: dict, metrics: dict) -> None:
+def log_hyperparams_tensorboard(writer: SummaryWriter, hyperparams: dict, metrics: dict) -> None:
     writer.add_hparams(hyperparams, metrics)
+
+
+def plot_tensorboard(
+    writer: SummaryWriter,
+    model: Module,
+    iteration: int,
+    plotting_functions: tuple[PlottingFunction],
+) -> None:
+    for pf in plotting_functions:
+        descr, fig = pf(model, iteration)
+        writer.add_figure(descr, fig, global_step=iteration)
+
+
+def log_model_tensorboard(
+    writer: SummaryWriter,
+    model: Module,
+    dataloader: Union[None, DataLoader, DictDataLoader],
+) -> None:
+    logger.warning("Model logging is not supported by tensorboard. No model will be logged.")
+
+
+def write_mlflow(writer: Any, loss: float | None, metrics: dict, iteration: int) -> None:
+    writer.log_metrics({"loss": float(loss)}, step=iteration)  # type: ignore
+    writer.log_metrics(metrics, step=iteration)  # logs the single metrics
+
+
+def log_hyperparams_mlflow(writer: Any, hyperparams: dict, metrics: dict) -> None:
+    writer.log_params(hyperparams)  # type: ignore
+
+
+def plot_mlflow(
+    writer: Any,
+    model: Module,
+    iteration: int,
+    plotting_functions: tuple[PlottingFunction],
+) -> None:
+    for pf in plotting_functions:
+        descr, fig = pf(model, iteration)
+        writer.log_figure(fig, descr)
+
+
+def log_model_mlflow(
+    writer: Any,
+    model: Module,
+    dataloader: Union[None, DataLoader, DictDataLoader],
+) -> None:
+    if dataloader is not None:
+        xs: InputData
+        xs, *_ = next(iter(dataloader))
+        preds = model(xs)
+        if isinstance(xs, Tensor):
+            xs = xs.numpy()
+            preds = preds.detach().numpy()
+        elif isinstance(xs, dict):
+            for key, val in xs.items():
+                xs[key] = val.numpy()
+            for key, val in preds.items():
+                preds[key] = val.detach.numpy()
+        signature = infer_signature(xs, preds)
+    else:
+        signature = None
+    writer.pytorch.log_model(model, artifact_path="model", signature=signature)
+
+
+TRACKER_MAPPING = {
+    ExperimentTrackingTool.TENSORBOARD: write_tensorboard,
+    ExperimentTrackingTool.MLFLOW: write_mlflow,
+}
+
+LOGGER_MAPPING = {
+    ExperimentTrackingTool.TENSORBOARD: log_hyperparams_tensorboard,
+    ExperimentTrackingTool.MLFLOW: log_hyperparams_mlflow,
+}
+
+PLOTTER_MAPPING = {
+    ExperimentTrackingTool.TENSORBOARD: plot_tensorboard,
+    ExperimentTrackingTool.MLFLOW: plot_mlflow,
+}
+
+MODEL_LOGGER_MAPPING = {
+    ExperimentTrackingTool.TENSORBOARD: log_model_tensorboard,
+    ExperimentTrackingTool.MLFLOW: log_model_mlflow,
+}
+
+
+def write_tracker(
+    args: Any, tracking_tool: ExperimentTrackingTool = ExperimentTrackingTool.TENSORBOARD
+) -> None:
```

Comment:
```suggestion
) -> dict:
```
Maybe defining the return type would help removing the `# type: ignore` directive ?

Reply:
Fixed by defined some types for the mappings in fdf05f1